### PR TITLE
Command descriptions

### PR DESCRIPTION
**Pull request by giteden**
## Proposed Changes

add argument descriptions/minor doc fixes, to the following commands:

- show
- create
- envs
- fork
- import
- export
- tag
- snap
- status
- new
- start
- scope
- lane
- test
- compile
- install
- use
- app

### CODE REVIEW

File: scopes/envs/envs/envs.cmd.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { EnvOverview } from './components/env-overview';
 
 export class ListEnvsCmd implements Command {
   name = 'list';
-  description = 'list all envs available in the workspace';
+  description = 'lists all envs available in the workspace';
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.cmd.tsx:13]: @giteden that's not correct.
it doesn't show available envs, but the list of components and their envs.

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/component/component/show/show.cmd.ts [hunk 2]
```diff
@@ -9,18 +9,19 @@ import LegacyShow from '@teambit/legacy/dist/cli/commands/public-cmds/show-cmd';
 import { ComponentMain } from '../component.main.runtime';
 
 export class ShowCmd implements Command {
-  name = 'show <id>';
-  description = 'show a component';
+  name = 'show <component-name>';
+  description = "Displays the component's essential information.";
```
Comment (davidfirst) [inline scopes/component/component/show/show.cmd.ts:13]: as far as I remember, in all other places the descriptions are not capitalized.

File: scopes/component/forking/fork.cmd.ts [hunk 3]
```diff
@@ -9,22 +9,38 @@ export type ForkOptions = {
...
+  description = 'EXPERIMENTAL. Creates a new component out of an existing one (copies source files and config).';
+  arguments = [
+    { name: 'source-component-id', description: 'the component id of the source component' },
+    {
+      name: 'target-component-name',
+      description: 'the name for the new component (component name without scope)',
+    },
+  ];
+
   extendedDescription = `note that [target-name] is the name only without the scope.
```
Comment (davidfirst) [inline scopes/component/forking/fork.cmd.ts:22]:
```
- [target-name]
+ [target-component-name]
```

File: scopes/defender/tester/test.cmd.tsx [hunk 4]
```diff
@@ -21,8 +21,15 @@ type TestFlags = {
...
 
 export class TestCmd implements Command {
-  name = 'test [pattern]';
-  description = 'test set of components in your workspace';
+  name = 'test [component-pattern]';
+  description = 'Tests components in your workspace';
+  arguments = [
+    {
+      name: 'component-pattern',
+      description:
```
Comment (davidfirst) [inline scopes/defender/tester/test.cmd.tsx:29]: (yes, it's very confusing and must be fixed) this "pattern" is the old pattern, which doesn't have the ability of using "!".  (that's why it didn't have the constant pattern variable as other commands).

File: scopes/lanes/lanes/lane.cmd.ts [hunk 5]
```diff
@@ -336,7 +336,8 @@ export class LaneMergeCmd implements Command {
 }
 
 export class LaneRemoveCmd implements Command {
-  name = 'remove <lane...>';
+  name = 'remove <lanes...>';
+  arguments: [{ name: 'lanes...'; description: 'A list of lane names, separated by spaces' }];
```
Comment (davidfirst) [inline scopes/lanes/lanes/lane.cmd.ts:340]: when an arg has the array form: `...`, in the CLI it is separated by a space. not a comma.

**Review by davidfirst — CHANGES_REQUESTED** [review]

File: scopes/envs/envs/envs.cmd.tsx (hunk 1 above)
Comment (giteden) [reply scopes/envs/envs/envs.cmd.tsx:13]: @GiladShoham you're talking about `bit envs`. `bit envs list` lists all the available envs.

File: scopes/component/forking/fork.cmd.ts (hunk 3 above)
Comment (giteden) [reply scopes/component/forking/fork.cmd.ts:22]: removed the extended description and added the --scope flag comment in the target-component-name arg description

File: scopes/component/snapping/snap-cmd.ts [hunk 6]
```diff
@@ -9,12 +9,20 @@ import { SnapResults } from '@teambit/legacy/dist/api/consumer/lib/snap';
 import { SnappingMain } from './snapping.main.runtime';
 
 export class SnapCmd implements Command {
-  name = 'snap [id]';
-  description = 'EXPERIMENTAL. record component changes';
+  name = 'snap [component_names...]';
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/component/snapping/snap-cmd.ts:12]: fixed

File: scopes/component/status/status-cmd.ts [hunk 7]
```diff
@@ -8,28 +8,23 @@ import { formatBitString, formatNewBit } from '@teambit/legacy/dist/cli/chalk-bo
...
 const trackDirDesc = `these components were added by an older version of Bit and therefore have "trackDir" record in the .bitmap file
 please run "bit migrate --harmony" to convert these records to "rootDir".`;
 
 export class StatusCmd implements Command {
   name = 'status';
-  description = 'show the working area component(s) status';
+  description = 'Presents the current status of components in the workspace, and notifies when issues are detected.';
   group = 'development';
-  extendedDescription = `https://${BASE_DOCS_DOMAIN}/workspace/workspace-status`;
+  extendedDescription: string;
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/component/status/status-cmd.ts:27]: removed

File: scopes/defender/tester/test.cmd.tsx (hunk 4 above)
Comment (giteden) [reply scopes/defender/tester/test.cmd.tsx:29]: fixed

File: scopes/envs/envs/envs.cmd.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/envs/envs/envs.cmd.tsx:13]: that's the convention for describing commands (see git, for example). I've done the same in all other commands.

File: scopes/lanes/lanes/lane.cmd.ts (hunk 5 above)
Comment (giteden) [reply scopes/lanes/lanes/lane.cmd.ts:340]: not sure what you mean (i wrote 'space', not 'comma')

File: scopes/scope/export/export-cmd.ts [hunk 8]
```diff
@@ -6,15 +6,28 @@ import { CURRENT_UPSTREAM, WILDCARD_HELP } from '@teambit/legacy/dist/constants'
 import GeneralError from '@teambit/legacy/dist/error/general-error';
 import chalk from 'chalk';
 import { isEmpty } from 'lodash';
+import { CommandArg } from '../../../dist/cli/command';
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/scope/export/export-cmd.ts:9]: done

File: scopes/workspace/workspace/envs-subcommands/envs-set.cmd.ts [hunk 9]
```diff
@@ -4,8 +4,34 @@ import { PATTERN_HELP } from '@teambit/legacy/dist/constants';
...
 export class EnvsSetCmd implements Command {
-  name = 'set <pattern> <env>';
-  description = 'set an environment for component(s)';
+  name = 'set <component-pattern> <env>';
+  description = 'sets one or more components with a development environment (env)';
+  arguments = [
+    {
+      name: 'component-pattern',
+      description:
+        'component name, component id, or component pattern. use component pattern to select multiple components. \nuse commas to separate patterns and "!" to exclude. e.g. "ui/**, !ui/button"\nwrap the pattern with quotes',
```
[the remark this answers is not in the document]
Comment (giteden) [reply scopes/workspace/workspace/envs-subcommands/envs-set.cmd.ts:13]: removed

File: scopes/lanes/lanes/lane.cmd.ts (hunk 5 above)
Comment (davidfirst) [reply scopes/lanes/lanes/lane.cmd.ts:340]: oops.. my bad

File: scopes/component/status/status-cmd.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (davidfirst) [reply scopes/component/status/status-cmd.ts:27]: I still see `extendedDescription: string;` you can remove it completely if not used.

**Comment by davidfirst** [comment]
Due to some issues with this branch, a new branch was created with the same commits:
https://github.com/teambit/bit/pull/5969

**Pull request closed without merging**